platform/view/services/db/driver/sql/common/pagination.go

```diff
@@ -85,25 +87,140 @@ func (p *OffsetPagination) GoBack(numOfpages int) (driver.Pagination, error) {
 func (p *OffsetPagination) Prev() (driver.Pagination, error) { return p.GoBack(1) }
 func (p *OffsetPagination) Next() (driver.Pagination, error) { return p.GoForward(1) }
 
+type KeysetPagination struct {
+	offset   int
+	pageSize int
+	// name of the field in the database that is a unique id of the records
+	sqlIdName string
```

Comment:
the type of this is `FieldName`. (only of this property. `idFieldName` is not. keep it as string.